src/fprime_extras/lint/rules/rule_valid_port_types.py

```diff
@@ -0,0 +1,162 @@
+import logging
+
+from ...core.hooks import HookRegistry as hooks
+from .base import AbstractRule
+from .base import LintSeverity
+from .base import Notification
+from .registry import RuleListRegistry
+from ...core.util import find_fprime
+
+from pathlib import Path
+import os, sys
+import lxml.etree as etree
+from lxml.etree import XMLSyntaxError
+
+log = logging.getLogger(__name__)
+slug = 'invalid-port-type'
+# TODO: import following list from model instead of defining it here.
+port_kinds = ['output', 'sync_input', 'async_input', 'guarded_input']
+
+
+@RuleListRegistry.register()
+class RuleDanglingPortConnections(AbstractRule):
```

Comment:
Update the class name to be ``ValidPortTypes`` or something similar, this one conflicts with an existing rule.

Reply:
(I realize this might have been my mistake, but your PR has brought it to light)

Reply:
I fixed this for this file and the port type rule file as well

Reply:
great!